decode.go

```diff
@@ -269,6 +269,39 @@ func errTypeMismatch(dstType, srcType reflect.Type) *typeError {
 	return &typeError{dstType: dstType, srcType: srcType}
 }
 
+type unknownFieldError struct {
+	err error
+}
+
+func (e *unknownFieldError) Error() string {
+	return e.err.Error()
+}
+
+func errUnknownField(msg string, tk *token.Token) *unknownFieldError {
+	return &unknownFieldError{err: errors.ErrSyntax(msg, tk)}
+}
+
+func (d *Decoder) deleteStructKeys(str reflect.Value, unknownFields map[string]ast.Node) error {
```

Comment:
'str' is often used to mean 'string' ...
Would you please fix to first argument name from `str` to `structValue`  ?